Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,367 @@
+        if (TryComp<KnockedDownComponent>(contained, out var _knockedDown)
+            || TryComp<MobStateComponent>(contained, out var mobStateComponent) && _mobStateSystem.IsIncapacitated(contained))
+        {
+            _standingStateSystem.Down(contained);
+        }
+        else
+        {
+            _standingStateSystem.Stand(contained);
+        }
+
+        UpdateAppearance(uid, cryoPodComponent);
+    }
+
+    #region Interaction
+
+    private void HandleDragDropOn(EntityUid uid, CryoPodComponent cryoPodComponent, DragDropEvent args)
+    {
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            return;
+        }
+
+        var doAfterArgs = new DoAfterEventArgs(args.User, cryoPodComponent.EntryDelay, default, args.Dragged)
+        {
+            BreakOnDamage = true,
+            BreakOnStun = true,
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            NeedHand = false,
+            BroadcastFinishedEvent = new DoInsertCryoPodEvent(cryoPodComponent, args.Dragged, uid),
+        };
+        _doAfterSystem.DoAfter(doAfterArgs);
+        args.Handled = true;
+    }
+
+    private void DoInsertCryoPod(DoInsertCryoPodEvent ev)
+    {
+        InsertBody(ev.Unit, ev.ToInsert, ev.CryoPod);
+    }
+
+    private void HandleInteractHand(EntityUid uid, CryoPodComponent cryoPodComponent, InteractHandEvent args)
+    {
+        if (!HasComp<ActiveCryoPodComponent>(uid))
+        {
+            return;
+        }
+        if (!EntityManager.TryGetComponent(args.User, out ActorComponent? actor))
+            return;
+
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            if (args.User == cryoPodComponent.BodyContainer.ContainedEntity)
+            {
+                return;
+            }
+            _userInterfaceSystem.TryOpen(uid, SharedHealthAnalyzerComponent.HealthAnalyzerUiKey.Key, actor.PlayerSession);
+            _userInterfaceSystem.TrySendUiMessage(
+                uid,
+                SharedHealthAnalyzerComponent.HealthAnalyzerUiKey.Key,
+                new SharedHealthAnalyzerComponent.HealthAnalyzerScannedUserMessage(cryoPodComponent.BodyContainer.ContainedEntity));
+            args.Handled = true;
+        }
+    }
+
+    private void OnInteractUsing(EntityUid uid, CryoPodComponent cryoPodComponent, InteractUsingEvent args)
+    {
+        if (args.Handled || !cryoPodComponent.Locked || cryoPodComponent.BodyContainer.ContainedEntity == null)
+            return;
+
+        if (EntityManager.TryGetComponent(args.Used, out ToolComponent? tool)
+            && tool.Qualities.Contains("Prying")) // Why aren't those enums?
+        {
+            if (cryoPodComponent.IsPrying)
+                return;
+            cryoPodComponent.IsPrying = true;
+
+            _toolSystem.UseTool(args.Used, args.User, uid, 0f,
```

Comment:
could convert this to use targeted events, instead? the form of these events lends itself towards that - see the UseTool function for more information

to summarize, you just have to define the doAfter event target, so that once the doafter finishes, it knows to send the event as a targeted event to the doAfter event target - and in this system, you'd just change it so that the event listener is instead listening to a targeted event, and you wouldn't have to add the uid of the cryopod and the reference to the cryopod component in the event